cumulusci/core/keychain/encrypted_file_project_keychain.py

```diff
@@ -197,34 +257,69 @@ def _load_org_files(self, dirname: str, constructor=None):
                     constructor(config) if constructor else config
                 )
 
-    def _set_org(self, org_config, global_org):
+    def _set_org(self, org_config, global_org, save=True):
+        if not global_org and not self.project_local_dir:
+            return
         if org_config.keychain:
             assert org_config.keychain is self
         assert org_config.global_org == global_org
         org_config.keychain = self
         org_config.global_org = global_org
-        encrypted = self._encrypt_config(org_config)
-        self._set_encrypted_org(org_config.name, encrypted, global_org)
 
-    def _set_encrypted_org(self, name, encrypted, global_org):
+        org_name = org_config.name
+
+        org_bytes = self._get_config_bytes(org_config)
+
+        if global_org:
+            org_config = GlobalOrg(org_bytes)
+        else:
+            org_config = LocalOrg(org_bytes)
+
+        self.orgs[org_name] = org_config
+
+        if save:
+            self._save_org(
+                org_name,
+                org_config.data,
+                global_org,
+            )
+
+    def _save_org(self, name, org_bytes, global_org):
+        """
+        @name - name of the org
+        @org_bytes - bytes-like objecte to write to disk
+        @global_org - whether or not this is a global org
+        """
         if global_org:
             filename = Path(f"{self.global_config_dir}/{name}.org")
         elif self.project_local_dir is None:
             return
         else:
             filename = Path(f"{self.project_local_dir}/{name}.org")
-        with open(filename, "wb") as f_org:
-            f_org.write(encrypted)
+
+        fd = os.open(filename, os.O_WRONLY | os.O_CREAT, SERVICE_ORG_FILE_MODE)
```

Comment:
https://docs.python.org/3/library/os.html#os.open says that we also need O_BINARY to open the file in binary mode on Windows